pattern/variable/ThingVariable.java

```diff
@@ -241,9 +241,10 @@ public String toString() {
         StringBuilder tail = new StringBuilder();
 
         if (reference().isName()) head.append(reference());
-        tail.append(Stream.of(relationConstraints, set(isaConstraint), hasConstraints, valueConstraints, isConstraints)
-                            .flatMap(Collection::stream).filter(Objects::nonNull).map(ThingConstraint::toString)
-                            .collect(Collectors.joining("" + COMMA + SPACE)));
+        tail.append(Stream.of(relationConstraints, isaConstraint == null ? new HashSet<IsaConstraint>() : set(isaConstraint),
+                              hasConstraints, valueConstraints, isConstraints)
+                              .flatMap(Collection::stream).filter(Objects::nonNull).map(ThingConstraint::toString)
+                              .collect(Collectors.joining("" + COMMA + SPACE)));
```

Comment:
if we're not sure we should put a TODO and mention the TODOs in an issue about toString